pkg/storage/uploader.go

```diff
@@ -203,3 +239,15 @@ func (u *BufferWriter) Reset() {
 func NewBufferWriter() *BufferWriter {
 	return &BufferWriter{buf: &bytes.Buffer{}}
 }
+
+// CreateUploader create multi upload request.
+func CreateUploader(ctx context.Context, s ExternalStorage, name string) (Uploader, error) {
+	switch storage := s.(type) {
+	case *S3Storage:
+		return storage.CreateUploader(ctx, name)
+	case *LocalStorage:
+		return storage.CreateUploader(ctx, name)
+	default:
+		return nil, errors.Annotatef(berrors.ErrStorageInvalidConfig, "unsupported externalStorage type %T", s)
+	}
+}
```

Comment:
why do we still need this function

Reply:
Because we still have the concept of `pure uploader` and `uploader with buffer cache` so I kept the `Uploader` interface. But the users don't need to `CreateUploader` manually anymore. Any advice?

Reply:
I think only CDC needs `CreateUploader`?

Reply:
After this PR no one will need `CreateUploader` anymore. If CDC wants to open a streaming file write, it should use `ExternalStorage.Open` and `ExternalFile.Write`.

But, we still have the concepts of `Uploader` and `UploaderWriter` but they are not exposed to users.

For `Uploader`, it will upload data to cloud storage and complete the upload when the file is finished. It will upload data directly without any compression or buffer.

`uploaderWriter` will wrap `Uploader` and buffer the data to make it satisfy the upload requirements (for example: s3 5M). `uploaderWriter` can also help buffer and compress the data to upload.

So I still keep this function.

Reply:
Open() doesn't create a writable file. You mean Create()? 

And this Uploader sounds like an implementation detail now. Why do we still need an abstract Uploader interface then? For instance, in Local FS and noop:// [*and GCS*](https://pkg.go.dev/cloud.google.com/go/storage#Writer) we don't need any chunking behavior and can skip the `UploaderWriter` entirely; while in S3 we automatically buffer by 5 MB and use the UploadPart API.

Reply:
Yes, I mean Create().